File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.rb
# @param {Integer[]} prices
# @return {Integer}
def get_descent_periods(prices)
    l = r = ans = 0
    while r < prices.length
        ans += r - l + 1
        r += 1
        if r < prices.length && prices[r] != prices[r-1] - 1
            l = r
        end
    end
    return ans
end

File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var getDescentPeriods = function(prices) {
    let [l, r, ans] = [0, 0, 0];
    while (r < prices.length) {
        ans += r - l + 1;
        r++;
        if (prices[r] !== prices[r - 1] - 1) {
            l = r;
        }
    }
    return ans;
};